pkg/api/install/install.go

```diff
@@ -34,35 +34,55 @@ import (
 	"k8s.io/kubernetes/pkg/runtime"
 )
 
-// userResources is a group of resources mostly used by a kubectl user
-var userResources = []string{"rc", "svc", "pods", "pvc"}
-
 const importPrefix = "k8s.io/kubernetes/pkg/api"
 
 var accessor = meta.NewAccessor()
 
+// availableVersions lists all known external versions for this group from most preferred to least preferred
+var availableVersions = []unversioned.GroupVersion{v1.SchemeGroupVersion}
```

Comment:
> Sounds good. This is in alignment with our goal of separating API machinery from API types. I think we should split the initialization into 2 phases: first step is registration, each install package register the group as available; second step is installation, each install package calls a function provided by the registered package to learn which group/version user wants to install, as specified in KUBE_API_VERSIONS, and if the environment variable is not set, then all registered group/version will be installed. I think the end result will be the init() in the regostered package will be removed, initialization is done in each group.
> 
> I'd like to implement the code if you agree with the design. Probably I should make an issue for this.

I like the direction, but I'm not as sure of the details.  I think I'd like to register everything as available and the selectively enable or disable particular versions after the fact by calling explicit methods.  That way I could choose to specify either a whitelist or a blacklist separate from using an environment variable.